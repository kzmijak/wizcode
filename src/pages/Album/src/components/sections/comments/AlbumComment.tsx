import { Avatar, Stack, Typography } from "@mui/joy";
import { AlbumCommentModel } from "models/AlbumComment";
import { FC } from "react";

export type AlbumCommentProps = Pick<
  AlbumCommentModel,
  "description" | "lastModificationDate" | "title" | "userAvatar" | "userName"
> & {
  orientation?: "row" | "column";
};

export const AlbumComment: FC<AlbumCommentProps> = ({
  description,
  lastModificationDate,
  title,
  userAvatar,
  userName,
  orientation,
}) => {
  const dateComponent = (
    <Typography level="body2">
      {lastModificationDate.toLocaleDateString("pl")}
    </Typography>
  );

  return (
    <Stack
      data-testid="comment-genuine"
      component="article"
      direction={orientation}
      spacing={3}
      padding={2}
      borderRadius={10}
      alignItems={orientation === "column" ? "center" : "start"}
      border={(theme) => `dotted 2px ${theme.palette.divider}`}
    >
      <Stack
        spacing={2}
        maxWidth={60}
        justifyContent="center"
        alignItems="center"
      >
        <Avatar
          variant="solid"
          color="warning"
          src={userAvatar}
          alt={`${userName}'s avatar`}
        />
        <Typography variant="solid" fontFamily="monospace">
          {userName}
        </Typography>
        {orientation === "column" && dateComponent}
      </Stack>
      <Stack flex={1}>
        <Typography level="h6">{title}</Typography>
        <Typography variant="soft" flex={1}>
          {description}
        </Typography>
      </Stack>
      {orientation === "row" && dateComponent}
    </Stack>
  );
};
