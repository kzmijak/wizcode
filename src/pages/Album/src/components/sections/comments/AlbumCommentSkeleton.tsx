import { Avatar, Box, Stack, styled } from "@mui/joy";
import { FC } from "react";

const Paragraph = styled(Box)({
  backgroundColor: "lightgray",
  height: "1em",
  minWidth: 70,
});

export const AlbumCommentSkeleton: FC = () => {
  return (
    <Stack
      data-testid="comment-skeleton"
      component="article"
      direction="row"
      spacing={3}
      padding={2}
      borderRadius={10}
      border={(theme) => `dotted 2px ${theme.palette.divider}`}
    >
      <Stack spacing={2} width={60} justifyContent="center" alignItems="center">
        <Avatar variant="solid" color="neutral" />
        <Paragraph />
      </Stack>
      <Stack flex={1} spacing={2}>
        <Paragraph />
        <Paragraph minHeight="3em" />
      </Stack>
      <Paragraph />
    </Stack>
  );
};
