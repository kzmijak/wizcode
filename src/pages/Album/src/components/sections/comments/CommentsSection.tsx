import { Divider, Typography } from "@mui/joy";
import { FC } from "react";
import { AlbumComment, AlbumCommentProps } from "./AlbumComment";
import {
  albumCommentsState,
  useAlbumCommentsFetch,
} from "modules/AlbumCommentsState";
import { useRecoilValue } from "recoil";
import { AlbumModel } from "models/Album";
import { AlbumCommentSkeleton } from "./AlbumCommentSkeleton";

type CommentsSectionProps = Pick<AlbumModel, "id"> &
  Pick<AlbumCommentProps, "orientation">;

export const CommentsSection: FC<CommentsSectionProps> = ({
  orientation,
  ...album
}) => {
  const { status } = useAlbumCommentsFetch(album.id);
  const albumComments = useRecoilValue(albumCommentsState(album.id));

  const skeleton = [...Array(8).keys()].map((key) => (
    <AlbumCommentSkeleton key={key} orientation={orientation} />
  ));
  const comments = albumComments.map((comment) => (
    <AlbumComment key={comment.id} {...comment} orientation={orientation} />
  ));

  return (
    <>
      <Typography level="h4">Comments</Typography>
      <Divider />
      {status === "loading" ? skeleton : comments}
    </>
  );
};
