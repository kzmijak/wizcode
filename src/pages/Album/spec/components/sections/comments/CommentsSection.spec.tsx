import { describe, it, expect, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import { CommentsSection } from "../../../../src/components/sections/comments/CommentsSection";
import { RecoilRoot } from "recoil";

const comments = [...Array(10).keys()].map((key) => ({
  id: key,
  userName: key,
  userAvatar: key,
  title: key,
  description: key,
  lastModificationDate: key,
}));

vi.mock("axios", () => ({
  get: vi.fn().mockImplementation(async () => {
    setTimeout(() => {
      Promise.resolve(comments);
    }, 1);
  }),
}));

describe("CommentsSection", () => {
  const Component = () => (
    <RecoilRoot>
      <CommentsSection id="some-album" />
    </RecoilRoot>
  );

  it("should match snapshot", () => {
    expect(render(<Component />)).toMatchSnapshot();
  });

  it("should initially display skeleton for comments", () => {
    render(<Component />);

    screen.queryAllByTestId("comment-skeleton").forEach((result) => {
      expect(result).toBeInTheDocument();
    });
  });

  it.skip("should display comments after data loaded", async () => {
    render(<Component />);

    expect(await screen.findAllByTestId("comment-genuine")).toBeInTheDocument();
    (await screen.findAllByTestId("comment-skeleton")).forEach((result) => {
      expect(result).toBeInTheDocument();
    });
  });
});
