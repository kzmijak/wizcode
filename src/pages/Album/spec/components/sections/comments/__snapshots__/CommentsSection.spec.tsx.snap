// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CommentsSection > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h4
        class="MuiTypography-root MuiTypography-h4 css-nk02qn-JoyTypography-root"
      >
        Comments
      </h4>
      <hr
        class="MuiDivider-root MuiDivider-horizontal css-10w8je6-JoyDivider-root"
      />
      <article
        class="MuiStack-root css-19myurg-JoyStack-root"
        data-testid="comment-skeleton"
      >
        <div
          class="MuiStack-root css-1fpcv8p-JoyStack-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-variantSolid MuiAvatar-colorNeutral MuiAvatar-sizeMd css-ayhc9u-JoyAvatar-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorInherit MuiSvgIcon-fontSizeXl MuiAvatar-fallback css-31hyst-JoySvgIcon-root-JoyAvatar-fallback"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <div
            class="MuiBox-root css-5xjxzk"
          />
        </div>
        <div
          class="MuiStack-root css-1bf8ut2-JoyStack-root"
        >
          <div
            class="MuiBox-root css-5xjxzk"
          />
          <div
            class="MuiBox-root css-1v04on9"
          />
        </div>
        <div
          class="MuiBox-root css-5xjxzk"
        />
      </article>
      <article
        class="MuiStack-root css-19myurg-JoyStack-root"
        data-testid="comment-skeleton"
      >
        <div
          class="MuiStack-root css-1fpcv8p-JoyStack-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-variantSolid MuiAvatar-colorNeutral MuiAvatar-sizeMd css-ayhc9u-JoyAvatar-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorInherit MuiSvgIcon-fontSizeXl MuiAvatar-fallback css-31hyst-JoySvgIcon-root-JoyAvatar-fallback"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <div
            class="MuiBox-root css-5xjxzk"
          />
        </div>
        <div
          class="MuiStack-root css-1bf8ut2-JoyStack-root"
        >
          <div
            class="MuiBox-root css-5xjxzk"
          />
          <div
            class="MuiBox-root css-1v04on9"
          />
        </div>
        <div
          class="MuiBox-root css-5xjxzk"
        />
      </article>
      <article
        class="MuiStack-root css-19myurg-JoyStack-root"
        data-testid="comment-skeleton"
      >
        <div
          class="MuiStack-root css-1fpcv8p-JoyStack-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-variantSolid MuiAvatar-colorNeutral MuiAvatar-sizeMd css-ayhc9u-JoyAvatar-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorInherit MuiSvgIcon-fontSizeXl MuiAvatar-fallback css-31hyst-JoySvgIcon-root-JoyAvatar-fallback"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <div
            class="MuiBox-root css-5xjxzk"
          />
        </div>
        <div
          class="MuiStack-root css-1bf8ut2-JoyStack-root"
        >
          <div
            class="MuiBox-root css-5xjxzk"
          />
          <div
            class="MuiBox-root css-1v04on9"
          />
        </div>
        <div
          class="MuiBox-root css-5xjxzk"
        />
      </article>
      <article
        class="MuiStack-root css-19myurg-JoyStack-root"
        data-testid="comment-skeleton"
      >
        <div
          class="MuiStack-root css-1fpcv8p-JoyStack-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-variantSolid MuiAvatar-colorNeutral MuiAvatar-sizeMd css-ayhc9u-JoyAvatar-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorInherit MuiSvgIcon-fontSizeXl MuiAvatar-fallback css-31hyst-JoySvgIcon-root-JoyAvatar-fallback"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <div
            class="MuiBox-root css-5xjxzk"
          />
        </div>
        <div
          class="MuiStack-root css-1bf8ut2-JoyStack-root"
        >
          <div
            class="MuiBox-root css-5xjxzk"
          />
          <div
            class="MuiBox-root css-1v04on9"
          />
        </div>
        <div
          class="MuiBox-root css-5xjxzk"
        />
      </article>
      <article
        class="MuiStack-root css-19myurg-JoyStack-root"
        data-testid="comment-skeleton"
      >
        <div
          class="MuiStack-root css-1fpcv8p-JoyStack-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-variantSolid MuiAvatar-colorNeutral MuiAvatar-sizeMd css-ayhc9u-JoyAvatar-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorInherit MuiSvgIcon-fontSizeXl MuiAvatar-fallback css-31hyst-JoySvgIcon-root-JoyAvatar-fallback"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <div
            class="MuiBox-root css-5xjxzk"
          />
        </div>
        <div
          class="MuiStack-root css-1bf8ut2-JoyStack-root"
        >
          <div
            class="MuiBox-root css-5xjxzk"
          />
          <div
            class="MuiBox-root css-1v04on9"
          />
        </div>
        <div
          class="MuiBox-root css-5xjxzk"
        />
      </article>
      <article
        class="MuiStack-root css-19myurg-JoyStack-root"
        data-testid="comment-skeleton"
      >
        <div
          class="MuiStack-root css-1fpcv8p-JoyStack-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-variantSolid MuiAvatar-colorNeutral MuiAvatar-sizeMd css-ayhc9u-JoyAvatar-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorInherit MuiSvgIcon-fontSizeXl MuiAvatar-fallback css-31hyst-JoySvgIcon-root-JoyAvatar-fallback"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <div
            class="MuiBox-root css-5xjxzk"
          />
        </div>
        <div
          class="MuiStack-root css-1bf8ut2-JoyStack-root"
        >
          <div
            class="MuiBox-root css-5xjxzk"
          />
          <div
            class="MuiBox-root css-1v04on9"
          />
        </div>
        <div
          class="MuiBox-root css-5xjxzk"
        />
      </article>
      <article
        class="MuiStack-root css-19myurg-JoyStack-root"
        data-testid="comment-skeleton"
      >
        <div
          class="MuiStack-root css-1fpcv8p-JoyStack-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-variantSolid MuiAvatar-colorNeutral MuiAvatar-sizeMd css-ayhc9u-JoyAvatar-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorInherit MuiSvgIcon-fontSizeXl MuiAvatar-fallback css-31hyst-JoySvgIcon-root-JoyAvatar-fallback"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <div
            class="MuiBox-root css-5xjxzk"
          />
        </div>
        <div
          class="MuiStack-root css-1bf8ut2-JoyStack-root"
        >
          <div
            class="MuiBox-root css-5xjxzk"
          />
          <div
            class="MuiBox-root css-1v04on9"
          />
        </div>
        <div
          class="MuiBox-root css-5xjxzk"
        />
      </article>
      <article
        class="MuiStack-root css-19myurg-JoyStack-root"
        data-testid="comment-skeleton"
      >
        <div
          class="MuiStack-root css-1fpcv8p-JoyStack-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-variantSolid MuiAvatar-colorNeutral MuiAvatar-sizeMd css-ayhc9u-JoyAvatar-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorInherit MuiSvgIcon-fontSizeXl MuiAvatar-fallback css-31hyst-JoySvgIcon-root-JoyAvatar-fallback"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <div
            class="MuiBox-root css-5xjxzk"
          />
        </div>
        <div
          class="MuiStack-root css-1bf8ut2-JoyStack-root"
        >
          <div
            class="MuiBox-root css-5xjxzk"
          />
          <div
            class="MuiBox-root css-1v04on9"
          />
        </div>
        <div
          class="MuiBox-root css-5xjxzk"
        />
      </article>
    </div>
  </body>,
  "container": <div>
    <h4
      class="MuiTypography-root MuiTypography-h4 css-nk02qn-JoyTypography-root"
    >
      Comments
    </h4>
    <hr
      class="MuiDivider-root MuiDivider-horizontal css-10w8je6-JoyDivider-root"
    />
    <article
      class="MuiStack-root css-19myurg-JoyStack-root"
      data-testid="comment-skeleton"
    >
      <div
        class="MuiStack-root css-1fpcv8p-JoyStack-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-variantSolid MuiAvatar-colorNeutral MuiAvatar-sizeMd css-ayhc9u-JoyAvatar-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorInherit MuiSvgIcon-fontSizeXl MuiAvatar-fallback css-31hyst-JoySvgIcon-root-JoyAvatar-fallback"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
        <div
          class="MuiBox-root css-5xjxzk"
        />
      </div>
      <div
        class="MuiStack-root css-1bf8ut2-JoyStack-root"
      >
        <div
          class="MuiBox-root css-5xjxzk"
        />
        <div
          class="MuiBox-root css-1v04on9"
        />
      </div>
      <div
        class="MuiBox-root css-5xjxzk"
      />
    </article>
    <article
      class="MuiStack-root css-19myurg-JoyStack-root"
      data-testid="comment-skeleton"
    >
      <div
        class="MuiStack-root css-1fpcv8p-JoyStack-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-variantSolid MuiAvatar-colorNeutral MuiAvatar-sizeMd css-ayhc9u-JoyAvatar-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorInherit MuiSvgIcon-fontSizeXl MuiAvatar-fallback css-31hyst-JoySvgIcon-root-JoyAvatar-fallback"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
        <div
          class="MuiBox-root css-5xjxzk"
        />
      </div>
      <div
        class="MuiStack-root css-1bf8ut2-JoyStack-root"
      >
        <div
          class="MuiBox-root css-5xjxzk"
        />
        <div
          class="MuiBox-root css-1v04on9"
        />
      </div>
      <div
        class="MuiBox-root css-5xjxzk"
      />
    </article>
    <article
      class="MuiStack-root css-19myurg-JoyStack-root"
      data-testid="comment-skeleton"
    >
      <div
        class="MuiStack-root css-1fpcv8p-JoyStack-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-variantSolid MuiAvatar-colorNeutral MuiAvatar-sizeMd css-ayhc9u-JoyAvatar-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorInherit MuiSvgIcon-fontSizeXl MuiAvatar-fallback css-31hyst-JoySvgIcon-root-JoyAvatar-fallback"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
        <div
          class="MuiBox-root css-5xjxzk"
        />
      </div>
      <div
        class="MuiStack-root css-1bf8ut2-JoyStack-root"
      >
        <div
          class="MuiBox-root css-5xjxzk"
        />
        <div
          class="MuiBox-root css-1v04on9"
        />
      </div>
      <div
        class="MuiBox-root css-5xjxzk"
      />
    </article>
    <article
      class="MuiStack-root css-19myurg-JoyStack-root"
      data-testid="comment-skeleton"
    >
      <div
        class="MuiStack-root css-1fpcv8p-JoyStack-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-variantSolid MuiAvatar-colorNeutral MuiAvatar-sizeMd css-ayhc9u-JoyAvatar-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorInherit MuiSvgIcon-fontSizeXl MuiAvatar-fallback css-31hyst-JoySvgIcon-root-JoyAvatar-fallback"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
        <div
          class="MuiBox-root css-5xjxzk"
        />
      </div>
      <div
        class="MuiStack-root css-1bf8ut2-JoyStack-root"
      >
        <div
          class="MuiBox-root css-5xjxzk"
        />
        <div
          class="MuiBox-root css-1v04on9"
        />
      </div>
      <div
        class="MuiBox-root css-5xjxzk"
      />
    </article>
    <article
      class="MuiStack-root css-19myurg-JoyStack-root"
      data-testid="comment-skeleton"
    >
      <div
        class="MuiStack-root css-1fpcv8p-JoyStack-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-variantSolid MuiAvatar-colorNeutral MuiAvatar-sizeMd css-ayhc9u-JoyAvatar-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorInherit MuiSvgIcon-fontSizeXl MuiAvatar-fallback css-31hyst-JoySvgIcon-root-JoyAvatar-fallback"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
        <div
          class="MuiBox-root css-5xjxzk"
        />
      </div>
      <div
        class="MuiStack-root css-1bf8ut2-JoyStack-root"
      >
        <div
          class="MuiBox-root css-5xjxzk"
        />
        <div
          class="MuiBox-root css-1v04on9"
        />
      </div>
      <div
        class="MuiBox-root css-5xjxzk"
      />
    </article>
    <article
      class="MuiStack-root css-19myurg-JoyStack-root"
      data-testid="comment-skeleton"
    >
      <div
        class="MuiStack-root css-1fpcv8p-JoyStack-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-variantSolid MuiAvatar-colorNeutral MuiAvatar-sizeMd css-ayhc9u-JoyAvatar-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorInherit MuiSvgIcon-fontSizeXl MuiAvatar-fallback css-31hyst-JoySvgIcon-root-JoyAvatar-fallback"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
        <div
          class="MuiBox-root css-5xjxzk"
        />
      </div>
      <div
        class="MuiStack-root css-1bf8ut2-JoyStack-root"
      >
        <div
          class="MuiBox-root css-5xjxzk"
        />
        <div
          class="MuiBox-root css-1v04on9"
        />
      </div>
      <div
        class="MuiBox-root css-5xjxzk"
      />
    </article>
    <article
      class="MuiStack-root css-19myurg-JoyStack-root"
      data-testid="comment-skeleton"
    >
      <div
        class="MuiStack-root css-1fpcv8p-JoyStack-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-variantSolid MuiAvatar-colorNeutral MuiAvatar-sizeMd css-ayhc9u-JoyAvatar-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorInherit MuiSvgIcon-fontSizeXl MuiAvatar-fallback css-31hyst-JoySvgIcon-root-JoyAvatar-fallback"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
        <div
          class="MuiBox-root css-5xjxzk"
        />
      </div>
      <div
        class="MuiStack-root css-1bf8ut2-JoyStack-root"
      >
        <div
          class="MuiBox-root css-5xjxzk"
        />
        <div
          class="MuiBox-root css-1v04on9"
        />
      </div>
      <div
        class="MuiBox-root css-5xjxzk"
      />
    </article>
    <article
      class="MuiStack-root css-19myurg-JoyStack-root"
      data-testid="comment-skeleton"
    >
      <div
        class="MuiStack-root css-1fpcv8p-JoyStack-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-variantSolid MuiAvatar-colorNeutral MuiAvatar-sizeMd css-ayhc9u-JoyAvatar-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorInherit MuiSvgIcon-fontSizeXl MuiAvatar-fallback css-31hyst-JoySvgIcon-root-JoyAvatar-fallback"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
        <div
          class="MuiBox-root css-5xjxzk"
        />
      </div>
      <div
        class="MuiStack-root css-1bf8ut2-JoyStack-root"
      >
        <div
          class="MuiBox-root css-5xjxzk"
        />
        <div
          class="MuiBox-root css-1v04on9"
        />
      </div>
      <div
        class="MuiBox-root css-5xjxzk"
      />
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
